src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -887,15 +883,23 @@ private bool Synchronize()
         {
             lock (this.lockObj)
             {
-                HashHeightPair tipToChase = this.TipToChase();
                 if (this.TipHashAndHeight == null)
                 {
                     this.logger.LogError("Failed to synchronise. Reason: {0} is null.", nameof(this.TipHashAndHeight));
                     this.logger.LogTrace("(-)[SYNCHRONIZED]:false");
                     return false;
                 }
 
-                if (tipToChase.Hash == this.TipHashAndHeight.HashBlock)
+                HashHeightPair federationWalletTip = this.federationWalletManager.LastBlockSyncedHashHeight();
+
+                if (this.chainIndexer.GetHeader(federationWalletTip.Hash) == null)
+                {
+                    // Indicates that the federation wallet's tip is not on chain.
+                    this.logger.LogTrace("(-)[SYNCHRONIZED]:true");
```

Comment:
This is misleading, we're not synchronized here. By returning true all of the methods are going to proceed doing things, which they really shouldn't as the CCTS and FWM are using incorrect information as they're on the wrong chain.

Reply:
OK yes I was hoping you would comment here as I was unsure of this. So basically we need to let `Synchronize` return false and then just let the caller do nothing if so?